fuel_test/deploy_env.sh

```diff
@@ -0,0 +1,287 @@
+#!/bin/bash
+
+set -x
+
+function wait_for_fm {
+	# Wait for fuel master booting and return its IP address
+	local xs_host="$1"
+	local fm_name="$2"
+	local retry_count=$3
+	local retry_interval=$4
+
+	local fm_ip
+	local counter=0
+	while [ $counter -lt $retry_count ]; do
+		fm_ip=$(ssh -qo StrictHostKeyChecking=no root@$xs_host \
+	'eth1_ip=$(xe vm-list name-label="'$fm_name'" params=networks --minimal | egrep -Eo "1/ip: ([0-9]+\.){3}[0-9]+");echo ${eth1_ip: 5}')
+		if [ -n "$fm_ip" ]; then
+			echo $fm_ip
+			return
+		fi
+		let counter=counter+1
+		sleep $retry_interval
+	done
+}
+
+function start_node {
+	# Boot up given node
+	local xs_host="$1"
+	local vm="$2"
+	ssh -qo StrictHostKeyChecking=no root@$xs_host \
+	'
+	set -x
+
+	vm="'$vm'"
+	xe vm-start vm=$vm
+	'
+}
+
+function build_plugin {
+	# Build plugin with given refspec
+	# refspec can be empty
+	local fm_ip="$1"
+	local refspec="$2"
+	ssh -qo StrictHostKeyChecking=no root@$fm_ip \
+	'
+set -x
+pip install virtualenv
+cd /root/
+virtualenv fuel-devops-venv
+. fuel-devops-venv/bin/activate
+pip install fuel-plugin-builder
+yum install git createrepo dpkg-devel dpkg-dev rpm rpm-build -y
+git clone https://review.openstack.org/openstack/fuel-plugin-xenserver
+cd fuel-plugin-xenserver
+git fetch https://review.openstack.org/openstack/fuel-plugin-xenserver '"$refspec"'
+git checkout FETCH_HEAD
+fpb --check .
+fpb --build .
+	'
+}
+
+function wait_for_nailgun {
+	# Wait for nailgun service started before plugin can be installed
+	local fm_ip="$1"
+	local retry_count=$2
+	local retry_interval=$3
+
+	local counter=0
+	local ready
+	while [ $counter -lt $retry_count ]; do
+		ready=$(ssh -qo StrictHostKeyChecking=no root@$fm_ip \
+		'
+		export FUELCLIENT_CUSTOM_SETTINGS="/etc/fuel/client/config.yaml"
+		fuel plugins &> /dev/null
```

Comment:
Perhaps add a comment stating why you need to wait for fuel plugins to not error